apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -651,4 +652,38 @@ export class ApplicationController {
 
     return application
   }
+
+  @Put('application/:id/presignedUrl')
+  @ApiParam({
+    name: 'id',
+    type: String,
+    required: true,
+    description: 'The id of the application to update the state for.',
+    allowEmptyValue: false,
+  })
+  @ApiOkResponse({ type: ApplicationResponseDto })
+  @UseInterceptors(ApplicationSerializer)
+  async getPresignedUrl(
+    @Param('id', new ParseUUIDPipe(), ApplicationByIdPipe)
+    application: Application,
+    @Body() input: PresignedUrlDto,
+  ): Promise<ApplicationResponseDto> {
+    this.fileService.validateApplicationType(application.typeId)
+
+    const { type } = input
+
+    const url = this.fileService.getPresignedUrl(application.id, type)
+
+    const { updatedApplication } = await this.applicationService.update(
+      application.id,
+      {
+        attachments: {
+          ...application.attachments,
+          [type]: url,
+        },
+      },
+    )
```

Comment:
Nitpick: Since we are destructuring the `input` we could also do that with the `application`

That being said I actually think that in this case the most readable code would be skipping the desctructuring of `input` so when reading the code it is clear where each prop is coming from.

Reply:
The way this controller is architectured is that it takes an application id uses an interceptor to look that id up in the db and returns that application to the function. 
So you are talking about doing:
`const { id, typeId } = application`?
Agree with the second point since in the scope of this small function we do actually have two types, pdf type and application type so it is nice to have clarity there, will fix 😄